src/components/dropdown/CustomRender.tsx

```diff
@@ -81,168 +69,127 @@ export const SingleValue = (props: any) => {
-          <TagRender {...rendererProps} />
-        </div>
-      ) : (
-        <components.MultiValue {...props}>
-          <div onMouseDown={stopPropagation} className="tk-px-1" style={{display:'flex'} }>
-            <div className="tk-tag tk-pr-1">
-              {props.data?.label}
-            </div>
-            <div>
-              <Icon
-                iconName="cross"
-                onClick={() => props.removeProps.onClick()}
-              />
-            </div>
+  return (<div>
+    {TagRender ? 
+      <div onMouseDown={stopPropagation}>
+        <TagRender {...rendererProps} />
+      </div>
+      : <components.MultiValue {...props}>
+        <div onMouseDown={stopPropagation} className="tk-px-1" style={{display:'flex'} }>
+          <div className="tk-tag tk-pr-1">
+            {props.data?.label}
           </div>
-        </components.MultiValue>
-      )}
-    </div>
-  );
-};
-
-export const MultiValueContainerOverride = ({ children, ...props }: any) => {
-  return (
-    <components.MultiValueContainer {...props}>
-      <div>{children}</div>
-    </components.MultiValueContainer>
-  );
+          <div>
+            <Icon
+              iconName="cross"
+              onClick={() => props.removeProps.onClick()}
+            />
+          </div>
+        </div>
+      </components.MultiValue>}
+  </div>);
 };
 
-export const MultiValueRemove = () => {
-  return null;
-};
+export const MultiValueContainerOverride = ({ children, ...props }: any) => 
+  <components.MultiValueContainer {...props}>
+    <div>{children}</div>
+  </components.MultiValueContainer>;
 
 /**
  * This component controls the behavior of the expandable arrow displayed on
  * the right side of the Dropdown component.
- *
- * Default:
- *    visible -> Simple select
- *    hidden -> Multiple Select (isMulti prop)
- *
+ * Default: visible -> Simple select | hidden -> Multiple Select (isMulti prop)
  * The displayArrowIndicator prop from the Dropdown can override it
  */
 export const DropdownIndicator = (props: any) => {
   const { isMulti, displayArrowIndicator, menuIsOpen } = props?.selectProps;
-  return (
-    <div>
-      {isMulti ? (
-        <div>
-          {displayArrowIndicator ? (
-            <components.DropdownIndicator {...props} />
-          ) : (
-            <components.DropdownIndicator {...props} className="tk-d-none" />
-          )}
-        </div>
-      ) : (
-        <div>
-          {displayArrowIndicator ? (
-            <components.DropdownIndicator {...props}>
-              <Icon
-                iconName={menuIsOpen ? 'drop-up' : 'drop-down'}
-                className="tk-select__single-value"
-              ></Icon>
-            </components.DropdownIndicator>
-          ) : (
-            <components.DropdownIndicator {...props} className="tk-d-none" />
-          )}
-        </div>
-      )}
-    </div>
-  );
+  return (<> {isMulti ? 
+    <>{displayArrowIndicator ? 
+      <components.DropdownIndicator {...props}/>
```

Comment:
too much duplication, we can play instead with classnames and children instead of repeating the whole thing for each condition
it's barely readable

Reply:
done